### ISSUE TITLE
[Bug]: dn cannot create a checkpoint for a long time

### ISSUE THREAD

**Issue by XuPeng-SH** [matrixorigin/matrixone#12225]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

DN cannot create a checkpoint for a long time. There will be a `blk` always marked as `dirty` and it will block the checkpoint generation.

**Comment by XuPeng-SH** [comment]
#12174

**Comment by aptend** [comment]
<img width="2263" alt="image" src="https://github.com/matrixorigin/matrixone/assets/49832303/a1a829ef-df44-426a-9be9-e75ca339130b">

目前 checkpoint 执行正常

**Comment by XuPeng-SH** [comment]
fixed by #12245
